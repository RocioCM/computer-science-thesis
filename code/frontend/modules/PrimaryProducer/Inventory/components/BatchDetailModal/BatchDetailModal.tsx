import Modal, { ModalProps } from '@/common/components/Modal';
import { useEffect, useState } from 'react';
import { BottleBatch } from '../../types';
import PrimaryProducerServices from '../../services';
import { toast } from 'react-toastify';
import LoadingSpinner from '@/common/components/LoadingSpinner';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
}

const BatchDetailModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  ...props
}) => {
  const [batch, setBatch] = useState<BottleBatch | null>(null);

  const fetchBatchData = async () => {
    if (!editingId) return;
    const { ok, data } = await PrimaryProducerServices.getBatch(editingId);
    if (ok) {
      setBatch(data);
    } else {
      toast.error('Ocurrió un error al cargar los datos del lote de botellas');
    }
  };

  useEffect(() => {
    if (editingId) fetchBatchData();
  }, [editingId]);

  return (
    <Modal handleCancel={handleCancel} big skippable={false} {...props}>
      <h2 className="w-full bg-n0">Lote de botellas #{editingId}</h2>

      {batch ? (
        <div className="h-full w-full overflow-auto hide-scroll relative pt-2">
          {/* Información General */}
          <div className="grid grid-cols-2 gap-3xl mt-4">
            <div className=" space-y-2">
              <h3 className="text-lg font-semibold border-b pb-2">
                Información General
              </h3>
              <p>
                <span className="font-semibold">Cantidad de envases:</span>{' '}
                {batch.quantity}
              </p>
              <p>
                <span className="font-semibold">Vendidos:</span>{' '}
                {batch.soldQuantity}
              </p>
              <p>
                <span className="font-semibold">Fecha de creación:</span>{' '}
                {new Date(batch.createdAt).toLocaleDateString()}
              </p>
              <p>
                <span className="font-semibold">Ubicación de origen:</span>{' '}
                {batch.bottleType.originLocation}
              </p>
            </div>
            {/* Detalles de la Botella */}
            <div>
              <h3 className="text-lg font-semibold border-b pb-2">
                Detalles de la Botella
              </h3>
              <div className="mt-2 space-y-2">
                <p>
                  <span className="font-semibold">Peso:</span>{' '}
                  {batch.bottleType.weight}g
                </p>
                <p>
                  <span className="font-semibold">Color:</span>{' '}
                  {batch.bottleType.color}
                </p>
                <p>
                  <span className="font-semibold">Espesor:</span>{' '}
                  {batch.bottleType.thickness} mm
                </p>
                <p>
                  <span className="font-semibold">Forma:</span>{' '}
                  {batch.bottleType.shapeType}
                </p>
              </div>
            </div>
          </div>

          {/* Composición */}
          <div className="mt-6">
            <h3 className="text-lg font-semibold border-b pb-2">Composición</h3>
            {batch.bottleType.composition.length > 0 ? (
              <table className="w-full mt-2 border border-gray-300 rounded-md">
                <thead>
                  <tr className="bg-gray-100 text-gray-700">
                    <th className="px-4 py-2 text-left">Material</th>
                    <th className="px-4 py-2 text-left">Cantidad</th>
                    <th className="px-4 py-2 text-left">Unidad</th>
                  </tr>
                </thead>
                <tbody>
                  {batch.bottleType.composition.map((material, index) => (
                    <tr key={index} className="border-t">
                      <td className="px-4 py-2">{material.name}</td>
                      <td className="px-4 py-2">{material.amount}</td>
                      <td className="px-4 py-2">{material.measureUnit}</td>
                    </tr>
                  ))}
                </tbody>
              </table>
            ) : (
              <p className="text-gray-600 mt-2">
                No hay información de composición.
              </p>
            )}
          </div>
          {/* Información Adicional */}
          {batch.bottleType.extraInfo && (
            <div className="mt-6">
              <h3 className="text-lg font-semibold border-b pb-2">
                Información Adicional
              </h3>
              <p className="text-gray-700 mt-2">{batch.bottleType.extraInfo}</p>
            </div>
          )}
        </div>
      ) : (
        <LoadingSpinner className="mt-8" />
      )}
    </Modal>
  );
};

export default BatchDetailModal;
