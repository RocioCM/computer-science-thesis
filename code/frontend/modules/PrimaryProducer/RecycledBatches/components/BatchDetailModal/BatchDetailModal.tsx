import Modal, { ModalProps } from '@/common/components/Modal';
import { useEffect, useState } from 'react';
import { RecyclingBatch } from '../../types';
import { toast } from 'react-toastify';
import LoadingSpinner from '@/common/components/LoadingSpinner';
import RecycledBatchesServices from '../../services';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
}

const BatchDetailModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  ...props
}) => {
  const [batch, setBatch] = useState<RecyclingBatch | null>(null);

  const fetchBatchData = async () => {
    if (!editingId) return;
    const { ok, data } = await RecycledBatchesServices.getRecyclingBatch(
      editingId
    );
    if (ok) {
      setBatch(data);
    } else {
      toast.error('Ocurrió un error al cargar los datos del lote de material');
    }
  };

  useEffect(() => {
    if (editingId) fetchBatchData();
  }, [editingId]);

  return (
    <Modal
      handleCancel={handleCancel}
      big
      skippable={false}
      {...props}
      className="flex !justify-end w-full h-full !items-end !p-0"
      contentClassName="!rounded-none animate__slideInRight h-screen !px-10"
    >
      <h2 className="w-full font-medium text-[1.75rem]  bg-n0  pb-2">
        Lote de material #{editingId}
      </h2>
      {batch ? (
        <div className="h-full w-full flex flex-col gap-4 overflow-auto hide-scroll relative pt-2">
          {/* Información General */}
          <section className="grid md:grid-cols-2 gap-2 border border-n2 rounded-xl p-4">
            <h3 className="text-lg font-semibold pb-2 col-span-2">
              Información General
            </h3>
            <p>
              <span className="font-semibold">Material:</span>{' '}
              {batch.materialType.includes('glass')
                ? `vidrio ${batch.materialType.replace('glass', '')}`
                : batch.materialType}
            </p>
            <p>
              <span className="font-semibold">Tamaño:</span>{' '}
              {batch.size.replace('bottles', 'botellas')}
            </p>
            <p>
              <span className="font-semibold">Peso:</span> {batch.weight} kg
            </p>
            <p>
              <span className="font-semibold">Fecha de creación:</span>{' '}
              {new Date(batch.createdAt).toLocaleDateString()}
            </p>
            {/* Composición */}
            <div className="col-span-2">
              <h3 className="text-lg font-semibold  pb-2">Composición</h3>
              {batch.composition.length > 0 ? (
                <table className="w-full mt-2 border border-gray-300 rounded-md">
                  <thead>
                    <tr className="bg-gray-100 text-gray-700">
                      <th className="px-4 py-2 text-left">Material</th>
                      <th className="px-4 py-2 text-left">Cantidad</th>
                      <th className="px-4 py-2 text-left">Unidad</th>
                    </tr>
                  </thead>
                  <tbody>
                    {batch.composition.map((material, index) => (
                      <tr key={index} className="border-t">
                        <td className="px-4 py-2">{material.name}</td>
                        <td className="px-4 py-2">{material.amount}</td>
                        <td className="px-4 py-2">{material.measureUnit}</td>
                      </tr>
                    ))}
                  </tbody>
                </table>
              ) : (
                <p className="text-gray-600 mt-2">
                  No hay información de composición.
                </p>
              )}
            </div>
          </section>

          {/* Información Adicional */}
          <section className="flex flex-col gap-2 border border-n2 rounded-xl p-4">
            {batch.extraInfo && (
              <div>
                <h3 className="text-lg font-semibold py-2">
                  Información Adicional
                </h3>
                <p className="text-gray-700 mt-2">{batch.extraInfo}</p>
                <hr />
              </div>
            )}
            {/* Botellas */}
            <div>
              <h3 className="text-lg font-semibold py-2">Botellas</h3>
              {batch.wasteBottleIds.length > 0 ? (
                <ul className="grid grid-cols-4 gap-2">
                  {batch.wasteBottleIds.map((bottleId) => (
                    <li
                      key={bottleId}
                      className="font-semibold border rounded-rl p-3"
                    >
                      Botella #{bottleId}
                    </li>
                  ))}
                </ul>
              ) : (
                <p className="text-gray-600 mt-2">No hay botellas asociadas.</p>
              )}
            </div>
          </section>
        </div>
      ) : (
        <div className="w-full h-full flex items-center justify-center">
          <LoadingSpinner className="mt-8" size="3rem" />
        </div>
      )}
    </Modal>
  );
};

export default BatchDetailModal;
