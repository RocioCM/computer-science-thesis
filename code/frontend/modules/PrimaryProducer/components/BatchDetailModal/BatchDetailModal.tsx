import Modal, { ModalProps } from '@/common/components/Modal';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
  handleSuccess: () => void;
}

const BatchDetailModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  handleSuccess,
  ...props
}) => {
  return (
    <Modal handleCancel={handleCancel} big skippable={false} {...props}>
      <h2 className="w-full bg-n0">Lote de botellas #{editingId}</h2>
    </Modal>
  );
};

export default BatchDetailModal;
