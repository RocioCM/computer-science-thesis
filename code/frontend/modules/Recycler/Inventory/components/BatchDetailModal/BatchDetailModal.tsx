import Modal, { ModalProps } from '@/common/components/Modal';
import { useEffect, useState } from 'react';
import { RecyclingBatch } from '../../types';
import RecyclerServices from '../../services';
import { toast } from 'react-toastify';
import LoadingSpinner from '@/common/components/LoadingSpinner';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
  handleSuccess: () => void;
}

const BatchDetailModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  ...props
}) => {
  const [batch, setBatch] = useState<RecyclingBatch | null>(null);

  const fetchBatchData = async () => {
    if (!editingId) return;
    const { ok, data } = await RecyclerServices.getRecyclingBatch(editingId);
    if (ok) {
      setBatch(data);
    } else {
      toast.error('Ocurrió un error al cargar los datos del lote de botellas');
    }
  };

  useEffect(() => {
    if (editingId) fetchBatchData();
  }, [editingId]);

  return (
    <Modal
      handleCancel={handleCancel}
      big
      skippable={false}
      {...props}
      className="flex !justify-end w-full h-full !items-end !p-0"
      contentClassName="!rounded-none animate__slideInRight h-screen !px-10"
    >
      <h2 className="w-full font-medium text-[1.75rem] bg-n0 pb-2 mb-1">
        Lote de botellas #{editingId}
      </h2>

      {batch ? (
        <div className="h-full w-full flex flex-col gap-4 overflow-auto hide-scroll relative">
          {/* Información General */}
          <section className="grid md:grid-cols-2 gap-2 border border-n2 rounded-xl p-4">
            <h3 className="text-lg font-semibold border-b py-2 col-span-2">
              Información General
            </h3>
            <p>
              <span className="font-semibold">Material:</span>{' '}
              {batch.materialType}
            </p>
            <p>
              <span className="font-semibold">Tamaño:</span> {batch.size}
            </p>
            <p>
              <span className="font-semibold">Peso:</span> {batch.weight} kg
            </p>
            <p>
              <span className="font-semibold">Fecha de creación:</span>{' '}
              {new Date(batch.createdAt).toLocaleDateString()}
            </p>
            {/* Composición */}
            <div className="mt-2 col-span-2">
              <h3 className="text-lg font-semibold py-2">Composición</h3>
              {batch.composition.length > 0 ? (
                <table className="w-full mt-2 border border-gray-300 rounded-md">
                  <thead>
                    <tr className="bg-gray-100 text-gray-700">
                      <th className="px-4 py-2 text-left">Material</th>
                      <th className="px-4 py-2 text-left">Cantidad</th>
                      <th className="px-4 py-2 text-left">Unidad</th>
                    </tr>
                  </thead>
                  <tbody>
                    {batch.composition.map((material, index) => (
                      <tr key={index} className="border-t">
                        <td className="px-4 py-2">{material.name}</td>
                        <td className="px-4 py-2">{material.amount}</td>
                        <td className="px-4 py-2">{material.measureUnit}</td>
                      </tr>
                    ))}
                  </tbody>
                </table>
              ) : (
                <p className="text-gray-600 mt-2">
                  No hay información de composición.
                </p>
              )}
            </div>
          </section>

          <section className="flex flex-col gap-2 border border-n2 rounded-xl p-4">
            {/* Información Adicional */}
            {batch.extraInfo && (
              <div className="mt-6">
                <h3 className="text-lg font-semibold border-t py-2">
                  Información Adicional
                </h3>
                <p className="text-gray-700 mt-2">{batch.extraInfo}</p>
              </div>
            )}

            {/* Botellas */}
            <div className="mt-2">
              <h3 className="text-lg font-semibold py-2">Botellas</h3>
              {batch.wasteBottleIds.length > 0 ? (
                <ul className="grid grid-cols-4 gap-2">
                  {batch.wasteBottleIds.map((bottleId) => (
                    <li
                      key={bottleId}
                      className="font-semibold border rounded-rl p-3"
                    >
                      Botella #{bottleId}
                    </li>
                  ))}
                </ul>
              ) : (
                <p className="text-gray-600 mt-2">No hay botellas asociadas.</p>
              )}
            </div>
          </section>
        </div>
      ) : (
        <div className="flex h-full w-full items-center justify-center">
          <LoadingSpinner className="mt-8" size="3rem" />
        </div>
      )}
    </Modal>
  );
};

export default BatchDetailModal;
