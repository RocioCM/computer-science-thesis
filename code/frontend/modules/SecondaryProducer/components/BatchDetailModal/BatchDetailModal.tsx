import Modal, { ModalProps } from '@/common/components/Modal';
import { useEffect, useState } from 'react';
import { BaseBottleBatch, ProductBottlesBatch } from '../../types';
import SecondaryProducerServices from '../../services';
import { toast } from 'react-toastify';
import LoadingSpinner from '@/common/components/LoadingSpinner';

interface Props extends ModalProps {
  handleCancel: () => any;
  editingId: number | null;
}

const BatchDetailModal: React.FC<Props> = ({
  editingId,
  handleCancel,
  ...props
}) => {
  const [batch, setBatch] = useState<ProductBottlesBatch | null>(null);
  const [baseBatch, setBaseBatch] = useState<BaseBottleBatch | null>(null);

  const fetchBatchData = async () => {
    if (!editingId) return;
    const { ok, data } = await SecondaryProducerServices.getBatch(editingId);
    if (ok) {
      setBatch(data);
      fetchBaseBatchData(data.originBaseBatchId);
    } else {
      toast.error('Ocurrió un error al cargar los datos del lote de envases');
    }
  };

  const fetchBaseBatchData = async (baseBatchId: number) => {
    if (!editingId) return;
    const { ok, data } = await SecondaryProducerServices.getBaseBatch(
      baseBatchId
    );
    if (ok) {
      setBaseBatch(data);
    } else {
      toast.error('Ocurrió un error al cargar los datos del lote base');
    }
  };

  useEffect(() => {
    if (editingId) fetchBatchData();
  }, [editingId]);

  return (
    <Modal
      handleCancel={handleCancel}
      big
      skippable={false}
      {...props}
      className="flex !justify-end w-full h-full !items-end !p-0"
      contentClassName="!rounded-none animate__slideInRight h-screen !px-10"
    >
      <h2 className="w-full font-medium text-[1.75rem] bg-n0  pb-2">
        Lote de envases #{editingId}
      </h2>

      {batch ? (
        <div className="h-full w-full overflow-auto hide-scroll relative pt-2">
          {/* Información General */}
          <section className="grid md:grid-cols-2 gap-2 border border-n2 rounded-xl p-4">
            <div className=" space-y-2">
              <h3 className="text-lg font-semibold border-b pb-2">
                Información General
              </h3>
              <p>
                <span className="font-semibold">Cantidad de envases:</span>{' '}
                {batch.quantity}
              </p>
              <p>
                <span className="font-semibold">Disponibles:</span>{' '}
                {batch.availableQuantity}
              </p>
              <p>
                <span className="font-semibold">Fecha de recepción:</span>{' '}
                {new Date(batch.createdAt).toLocaleDateString()}
              </p>
            </div>
            {/* Detalles de la Botella */}
            {baseBatch && (
              <div>
                <h3 className="text-lg font-semibold border-b pb-2">
                  Detalles de la Botella
                </h3>
                <div className="mt-2 space-y-2">
                  <p>
                    <span className="font-semibold">Peso:</span>{' '}
                    {baseBatch.bottleType.weight}g
                  </p>
                  <p>
                    <span className="font-semibold">Color:</span>{' '}
                    {baseBatch.bottleType.color}
                  </p>
                  <p>
                    <span className="font-semibold">Espesor:</span>{' '}
                    {baseBatch.bottleType.thickness} mm
                  </p>
                  <p>
                    <span className="font-semibold">Forma:</span>{' '}
                    {baseBatch.bottleType.shapeType}
                  </p>
                </div>
              </div>
            )}
            {/* Composición */}
            <div className="mt-2 col-span-2">
              <h3 className="text-lg font-semibold border-b pb-2">
                Composición
              </h3>
              {baseBatch?.bottleType.composition.length ? (
                <table className="w-full mt-2 border border-gray-300 rounded-md">
                  <thead>
                    <tr className="bg-gray-100 text-gray-700">
                      <th className="px-4 py-2 text-left">Material</th>
                      <th className="px-4 py-2 text-left">Cantidad</th>
                      <th className="px-4 py-2 text-left">Unidad</th>
                    </tr>
                  </thead>
                  <tbody>
                    {baseBatch.bottleType.composition.map((material, index) => (
                      <tr key={index} className="border-t">
                        <td className="px-4 py-2">{material.name}</td>
                        <td className="px-4 py-2">{material.amount}</td>
                        <td className="px-4 py-2">{material.measureUnit}</td>
                      </tr>
                    ))}
                  </tbody>
                </table>
              ) : (
                <p className="text-gray-600 mt-2">
                  No hay información de composición.
                </p>
              )}
            </div>
          </section>
        </div>
      ) : (
        <div className="w-full h-full flex items-center justify-center">
          <LoadingSpinner className="mt-8" />
        </div>
      )}
    </Modal>
  );
};

export default BatchDetailModal;
