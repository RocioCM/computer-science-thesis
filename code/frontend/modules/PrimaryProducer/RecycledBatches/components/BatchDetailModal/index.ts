import BatchDetailModal from './BatchDetailModal';

export default BatchDetailModal;
