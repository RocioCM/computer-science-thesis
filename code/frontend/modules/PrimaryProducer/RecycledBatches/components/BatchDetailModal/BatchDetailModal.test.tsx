import { render, screen, fireEvent, act } from '@testing-library/react';
import BatchDetailModal from './BatchDetailModal';

describe('Primary Producer Recycled Batches - BatchDetailModal', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  });

  afterEach(() => {
    const modalRoot = document.getElementById('modal-root');
    if (modalRoot) {
      document.body.removeChild(modalRoot);
    }
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <BatchDetailModal editingId={1} handleCancel={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });
});
