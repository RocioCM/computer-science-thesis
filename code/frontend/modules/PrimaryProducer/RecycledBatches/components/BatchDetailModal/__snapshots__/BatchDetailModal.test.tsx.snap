// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Producer Recycled Batches - BatchDetailModal Matches snapshot 1`] = `<div />`;
